applications/FluidDynamicsApplication/custom_processes/integration_point_statistics_process.h

```diff
@@ -0,0 +1,606 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Jordi Cotela
+//
+
+#if !defined(KRATOS_INTEGRATION_POINT_STATISTICS_PROCESS_H_INCLUDED)
+#define KRATOS_INTEGRATION_POINT_STATISTICS_PROCESS_H_INCLUDED
+
+// System includes
+#include <string>
+#include <iostream>
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "includes/kratos_parameters.h"
+#include "processes/process.h"
+#include "custom_utilities/statistics_record.h"
+
+namespace Kratos
+{
+///@addtogroup FluidDynamicsApplication
+///@{
+
+///@name Kratos Classes
+///@{
+
+/// Helper process to record statistics on the integration points of the mesh.
+class IntegrationPointStatisticsProcess: public Process
+{
+public:
+///@name Type Definitions
+///@{
+
+/// Pointer definition of IntegrationPointStatisticsProcess
+KRATOS_CLASS_POINTER_DEFINITION(IntegrationPointStatisticsProcess);
+
+typedef std::map< std::string, StatisticsSampler::Pointer > StatisticsDictionary;
+
+///@}
+///@name Life Cycle
+///@{
+
+/// Constructor using a ModelPart.
+/** @param rModelPart ModelPart the statistics will be calculated on.
+ *  @param pParameters Parameters object defining the desired statistics.
+ */
+IntegrationPointStatisticsProcess(ModelPart& rModelPart, Kratos::Parameters::Pointer pParameters):
+    Process(),
+    mrModelPart(rModelPart),
+    mpParameters(pParameters),
+    mDimension(3),
```

Comment:
Just wondering, this only works for 3D cases?

Reply:
Now I see, it is intended to compute the turbulent statistics...

Reply:
It should work in 2D too (the test is 2D) but I decided to default it to 3D since my main focus is in turbulent statistics, as you say. The point of the `mDimension` argument is to know exactly how many numbers I need to store for each quantity. For vectors it is only 2 or 3 numbers, but a vector co-variance requires 4 or 9 doubles (8 or 15 if you count that the variance update formula also requires storing the average). Given that this is stored per gauss point, I thought it would make a difference in memory usage.

Reply:
Also, the dimension is passed through the json input. I see now that it might make sense to pass it in the constructor, but I didn't consider it because there is no code that is really different in either case (it just changes how much memory is allocated/used).

Reply:
@jcotela thanks for the explanation!